### PR TITLE
indexing documentation

### PR DESCRIPTION
**Pull request by evanchooly**
@jyemin @rozza

Added a new page covering index configuration

### CODE REVIEW

File: docs/reference/content/guides/indexing.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  pre = "<i class='fa fa-file-text-o'></i>"
++++
+
+# Indexing
+
+Morphia provides annotations that allow developers to define indexes for a collection to be defined alongside the other mapping data on 
+an entity's source.  In addition to the familiar ascending/descending index types, Morphia and MongoDB support [TTL]({{< docsref 
+"core/index-ttl/" >}}), [text]({{< docsref "core/index-text/" >}}), and [geospatial]({{< docsref "applications/geospatial-indexes/" >}})
+ indexes.   When defining [text]({{< ref "#text-indexing" >}}) indexes there are certain restrictions which will be covered below.  Full
+  details for all these types are available in the manual.
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: "in the manual" : link to the manual

File: docs/reference/content/guides/indexing.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+arbitrary number of `@Field`s can be given but at least one must be present.
+
+#### type()
+*Default: IndexType.ASC*
+     
+Indicates the "type" of the index (ascending, descending, geo2D, geo2d sphere, or text) to create on the field.
+  
+*See [IndexType](/javadoc/org/mongodb/morphia/utils/IndexType.html)*
+     
+#### value()
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: Move this up above type() as it comes first in the example.  Make it clear that "fields"/"field3" is the value().

File: docs/reference/content/guides/indexing.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+{{% note class="important" %}}
+As of MongoDB version 3.0, the dropDups option is no longer available.
+{{% /note %}}
+
+#### expireAfterSeconds()
+*Optional* 
+
+Specifies a value, in seconds, as a [TTL]({{< docsref "core/index-ttl/" >}}) to control how long MongoDB retains documents in 
+this collection.  The field listed must be a date field.
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: Perhaps: The field listed must contain values that are dates.

Comment (evanchooly) [inline docs/reference/content/guides/indexing.md]: done.

File: docs/reference/content/guides/indexing.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+applied at the field level, the index is created using only that field and so the [`@Field`]({{< ref "guides/annotations.md#field" >}}) 
+annotations are unnecessary.  The options for the index are the same as defined [above]({{< ref "#options" >}}).  A field level index 
+definition would look like this:
+
+```java
+@Entity
+private class FieldIndex {
+    @Id
+    private ObjectId id;
+    @Indexed
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: Provide an example of setting some properties on @Indexed.

File: docs/reference/content/guides/indexing.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+Specifies the weight to use when creating a text index.  This value only makes sense when direction is `IndexType.TEXT`.
+
+### Options
+
+Options for an index are defined on the [`@IndexOptions`]({{< ref "guides/annotations.md#indexoptions" >}}).  More complete coverage can 
+be found in the [manual]({{< docsref "reference/method/db.collection.createIndex/#options" >}}) but we'll provide some basic coverage 
+here as well.
+
+#### background()
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: Show an example of specifying at least one option

File: docs/reference/content/guides/indexing.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    private ObjectId id;
+    @Indexed
+    private String name;
+    private String color;
+}
+```
+
+## Text Indexing
+
+Morphia's indexing supports MongoDB's text indexing and search functionality as we've briefly seen above.  Full details can be found on
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: "in the manual"

File: docs/reference/content/guides/indexing.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    @Indexed
+    private String name;
+    private String color;
+}
+```
+
+## Text Indexing
+
+Morphia's indexing supports MongoDB's text indexing and search functionality as we've briefly seen above.  Full details can be found on 
+the [manual]({{< docsref "core/index-text/" >}}) but there are few Morphia specific details to cover.  Indexed field names are validated
```
Comment (jyemin) [inline docs/reference/content/guides/indexing.md]: "there are a few"

**Pull request closed without merging**